src/main/host/thread_ptrace.c

```diff
@@ -1079,41 +1114,27 @@ bool threadptrace_isRunning(Thread* base) {
     return false;
 }
 
-void threadptrace_terminate(Thread* base) {
-    ThreadPtrace* thread = _threadToThreadPtrace(base);
-
-    /* make sure we cleanup circular refs */
-    if (thread->sys) {
-        syscallhandler_unref(thread->sys);
-        thread->sys = NULL;
-    }
-
-    if (!threadptrace_isRunning(base)) {
-        return;
-    }
-
-    // need to kill() and not ptrace() since the process may not be stopped
-    if (kill(thread->base.nativeTid, SIGKILL) < 0) {
-        warning("kill %d: %s", thread->base.nativePid, g_strerror(errno));
-    }
+static void _threadptrace_reapProcess(ThreadPtrace* thread) {
+    info("Reaping process %d now", thread->base.nativePid);
 
-    // Wait for the exit ptrace-stop. Because the shadow process is the natural
-    // parent of the child (even if spawned from a task/thread other than this
-    // one), this also reaps the child.
-    int wstatus;
-    if (_waitpid_spin(thread->base.nativePid, &wstatus, 0) < 0) {
-        error("waitpid: %s", g_strerror(errno));
-        return;
+    int wstatus = 0;
+    if (waitpid(thread->base.nativePid, &wstatus, __WALL) < 0) {
+        error("waitpid(%d) error %d: %s", thread->base.nativePid, errno, g_strerror(errno));
     }
 
     StopReason reason = _getStopReason(wstatus);
+    _threadptrace_updateChildState(thread, reason);
 
-    if (reason.type != STOPREASON_EXITED_NORMAL && reason.type != STOPREASON_EXITED_SIGNAL) {
-        error("Expected process %d to exit after SIGKILL, instead received status %d",
-              thread->base.nativePid, wstatus);
+    if (threadptrace_isRunning(&thread->base)) {
+        warning("Process %d to have exited, instead received status %d", thread->base.nativePid,
```

Comment:
I think you're missing a verb, e.g., "_Expected_ process %d to have exited"?

Reply:
This is actually dead code; deleted